psi4/driver/driver.py

```diff
@@ -873,6 +873,150 @@ def properties(*args, **kwargs):
         return core.variable('CURRENT ENERGY')
 
 
+def optimize_geometric(name, **kwargs):
+
+    # will move imports to top of file when i'm sure where this method goes
+    import qcengine as qcng
+    import qcelemental as qcel
+    import geometric
+
+    if '/' in name:
+        raise ValidationError(f'Composite methods like {name} are not yet supported with geomeTRIC')
+                                 
+    # Collect all user-specified keywords
+    keywords = {}
+    for opt in core.get_global_option_list():
+        if core.has_global_option_changed(opt):
+            keywords[opt] = core.get_global_option(opt)
+    basis = keywords['BASIS']
+
+    # Make sure the molecule the user provided is the active one
+    molecule = kwargs.get('molecule', core.get_active_molecule())
+
+    # If we are freezing cartesian, do not orient or COM
+    if core.get_local_option("OPTKING", "FROZEN_CARTESIAN"):
+        molecule.fix_orientation(True)
+        molecule.fix_com(True)
+    molecule.update_geometry()
+
+    input_data = {
+        "keywords": {
+            "convergence_set": core.get_global_option("G_CONVERGENCE"),
+            "maxiter": core.get_global_option("GEOM_MAXITER"),
+            "coordsys": "tric",
+            "enforce": 0.1,
+            "constraints": {},
```

Comment:
Constraints are probably the #1 most vital item here.

Reply:
Excellent, Lee-Ping does have a language: https://github.com/leeping/geomeTRIC/blob/master/examples/constraints.txt . Only other one I know was Gaussian's ModRedundant ("Options" tab off https://gaussian.com/opt/). Psi4 has a not-great-one due to constraints of the options class - https://github.com/psi4/psi4/blob/master/tests/opt-freeze-coords/input.dat#L21-L22 - and I know Rollin's wanting to get away from it.

Reply:
I would use the JSON version and not the string versions.

```python
{
    "constraints": {
        "set": [{"indices": [0, 1], "type": "distance", "value": 1.4}],
        "freeze": [{"indices": [0, 1, 2], "type": "angle"}],
    }
}
```

Reply:
You can now specify constraints in the JSON format (which geometric internally converts to strings). I added a pytest demonstrating how to do this.